### PR TITLE
ref(slack): verify signature with SDK class

### PR DESCRIPTION
**Pull request by cathteng**
Use `SignatureVerifier` from the `slack_sdk` to verify Slack webhook signatures.

### CODE REVIEW

File: src/sentry/integrations/slack/requests/base.py [hunk 1]
```diff
@@ -197,8 +210,40 @@ def authorize(self) -> None:
         verification_token = options.get("slack.verification-token")
 
         if signing_secret:
-            if self._check_signing_secret(signing_secret):
-                return
+            if self._organization and features.has(
```
Comment (ykamo001) [inline src/sentry/integrations/slack/requests/base.py:213]: any chance we can move stuff into smaller functions?

Comment (cathteng) [reply src/sentry/integrations/slack/requests/base.py]: i refactored it a bit

**Review by ykamo001 — APPROVED** [review]

File: src/sentry/integrations/slack/requests/base.py [hunk 2]
```diff
@@ -108,6 +112,15 @@ def _get_context(self):
         self._identity = context.identity
         self._user = context.user
 
+        if context.integration:
+            org_integrations = integration_service.get_organization_integrations(
+                integration_id=context.integration.id, status=0
+            )
+            if org_integrations:
+                self._organization = organization_service.get_org_by_id(
+                    id=org_integrations[0].organization_id
```
[the remark this answers is not in the document]
Comment (cathteng) [reply src/sentry/integrations/slack/requests/base.py:121]: doesn't the `if` fail if it's an empty array?

[the remark this answers is not in the document]
Comment (ykamo001) [reply src/sentry/integrations/slack/requests/base.py:121]: yeah true, sorry meant to type length check

[the remark this answers is not in the document]
Comment (cathteng) [reply src/sentry/integrations/slack/requests/base.py:121]: or do you mean on the result of `organization_service.get_org_by_id`

**Pull request merged**